integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala

```diff
@@ -851,13 +851,21 @@ object CommonUtil {
   def mergeIndexFiles(sparkContext: SparkContext,
       segmentIds: Seq[String],
       tablePath: String,
-      carbonTable: CarbonTable): Unit = {
-    if (CarbonProperties.getInstance().getProperty(
-      CarbonCommonConstants.CARBON_MERGE_INDEX_IN_SEGMENT,
-      CarbonCommonConstants.CARBON_MERGE_INDEX_IN_SEGMENT_DEFAULT).toBoolean) {
-      new CarbonMergeFilesRDD(sparkContext, AbsoluteTableIdentifier.from(tablePath,
-        carbonTable.getDatabaseName, carbonTable.getFactTableName).getTablePath,
-        segmentIds).collect()
+      carbonTable: CarbonTable,
+      mergeIndexProperty: Option[Boolean]): Unit = {
+    mergeIndexProperty match {
+      case Some(true) =>
+        new CarbonMergeFilesRDD(sparkContext, AbsoluteTableIdentifier.from(tablePath,
+          carbonTable.getDatabaseName, carbonTable.getFactTableName).getTablePath,
+          segmentIds).collect()
+      case _ =>
+        if (CarbonProperties.getInstance().getProperty(
+          CarbonCommonConstants.CARBON_MERGE_INDEX_IN_SEGMENT,
+          CarbonCommonConstants.CARBON_MERGE_INDEX_IN_SEGMENT_DEFAULT).toBoolean) {
```

Comment:
Add the validation for boolean in case user pass wrong boolean parameter then take default